tests/framework/client/client.go

```diff
@@ -0,0 +1,70 @@
+package client
+
+import (
+	"sync"
+
+	migrationsv1 "kubevirt.io/client-go/generated/kubevirt/clientset/versioned/typed/migrations/v1alpha1"
+
+	"kubevirt.io/kubevirt/tests/util"
+
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	"kubevirt.io/client-go/kubecli"
+)
+
+// TestClient
+//This client is a wrapped kubevirt client that exposes all client functionality
+//and can be used as a normal client. It provides a mechanism that track
+//non-namespaced resources created or updated during e2e tests and functions to rollback them at the end of the execution.
+
+type TestClient struct {
+	kubecli.KubevirtClient
+}
+
+type CleanableResource interface {
+	Clean()
+}
+
+var (
+	coreOnce                  sync.Once
+	testCoreV1Client          *TestCoreV1Client
```

Comment:
In the client we initialize the `coreV1Client` during [https://github.com/kubevirt/kubevirt/blob/a6a84d0388b12d8c812e19cb7cc88f5193e42bdf/staging/src/kubevirt.io/client-go/kubecli/kubecli.go#L146](https://github.com/kubevirt/kubevirt/blob/a6a84d0388b12d8c812e19cb7cc88f5193e42bdf/staging/src/kubevirt.io/client-go/kubecli/kubecli.go#L146) and it is done only once (because `CoreV1()` is managed by k8s itself)
I don't think it is necessary do it in the client itself, because it is related to the fact that I have to override the `CoreV1()` function. I don't know if this reply to your question.